packages/common/tests/conftest.py

```diff
@@ -1,109 +1,130 @@
 
+@pytest.fixture(scope="function")
+def docker_gdal_image():
+    """The GDAL docker image to use for the tests"""
+    return "ghcr.io/osgeo/gdal:alpine-small-latest"
 
-@pytest.fixture(scope="session", autouse=True)
-def setenv():
-    os.environ["DAGSTER_DEPLOYMENT"] = "pytest"
 
+@pytest.fixture(scope="function")
+def wkt_testarea():
+    """A small test area in the oldtown of Utrecht, incl. the Oudegracht."""
+    yield "Polygon ((136251.531 456118.126, 136620.128 456118.126, 136620.128 456522.218, 136251.531 456522.218, 136251.531 456118.126))"
 
-@pytest.fixture(scope="session", autouse=False)
-def database():
-    """Database for testing, including test data. Runs in a docker container."""
-    image_name = "balazsdukai/3dbag-sample-data:latest"
-    container_name = "pytest-3dbag-pipeline"
-    port = 5565
-
-    # temp dir that is mounted on the container for data
-    r = ''.join(choice(string.ascii_letters) for _ in range(8))
-    tmp = Path(f"/tmp/pytest-3dbag-{r}")
-    tmp.mkdir()
-    tmp.chmod(mode=0o777)
-
-    # set up
-    client = docker.from_env()
-    try:
-        image = client.images.get(image_name)
-    except errors.ImageNotFound:
-        image = client.images.pull(image_name)
-    try:
-        cont = client.containers.run(
-            image, name=container_name, detach=True, volumes=[f"{tmp}:/data", ],
-            ports={"5432/tcp": port})
-    except errors.DockerException as e:
-        try:
-            cont = client.containers.get(container_name)
-        except errors.ContextNotFound:
-            pytest.fail(f"Could not start docker container from {image_name}:\n{e}")
-        else:
-            cont.remove(force=True)
-            pytest.fail(f"Could not start docker container from {image_name}:\n{e}")
-
-    # check setup
-    print(f"Mounted {image_name}/data onto {tmp}")
-    print(f"{tmp} contents:")
-    for i in tmp.iterdir():
-        print(i)
-
-    # run tests, but wait for database to start first
-    sleep(8)
-    db = DatabaseConnection(user="db3dbag_user", password="db3dbag_1234",
-                            host="localhost", port=port, dbname="baseregisters")
-    yield tmp, db
-
-    # teardown
-    cont.remove(force=True, v=True)
-    rmtree(str(tmp))
-    print(f"Deleted docker container {container_name}")
-    print(f"Deleted local directory {tmp}")
 
+@pytest.fixture
+def database(postgresql):
+    db = DatabaseConnection(conn=postgresql)
+    PostgresFunctions(db)
+    tbl = PostgresTableIdentifier("public", "existing_table")
+    query = SQL(
+        """CREATE TABLE {table} (id INTEGER, value TEXT);
+                   INSERT INTO {table} VALUES (1, 'bla');
+                   INSERT INTO {table} VALUES (2, 'foo');"""
+    ).format(table=Identifier(tbl.schema.str, tbl.table.str))
+
+    db.send_query(query)
+    yield db
+    db.conn.rollback()
+
+
+@pytest.fixture
+def context(database, docker_gdal_image, wkt_testarea, tmp_path):
```

Comment:
In both conftests, I have moved code around because I wanted to make new fixtures which depend on other fixtures.

**Important:**
1.  Instead of creating the dagster context inside each test I have created context fixtures which are used in most tests.
2. There are now 2 database fixtures and 2 context fixtures. This has to do with how the database connection is used. Wherever I can, I use a mock-connection which is cleaned after the tests are done. However, wherever gdal  and bash commands are  involved, the connection expects an existing db and I cannot use the mock one. So I have 2 different database fixtures which I use depending on the kind of test.

Reply:
In this case, what does a mock-connection provide that an existing db in a docker container cannot?
In other words, if we already need to use an existing db, could we use that for all tests, instead of a mock-connection? So that there is only one database for all the tests, instead of two.

Reply:
I agree, let's discuss this in the meeting.